cdap-standalone/src/main/java/co/cask/cdap/StandaloneMain.java

```diff
@@ -386,6 +402,7 @@ public static void main(String[] args) {
         System.err.println("Failed to start Standalone CDAP");
         e.printStackTrace(System.err);
       }
+      main.stopExternalProcesses();
```

Comment:
Hmm not quite sure. I guess the earlier logic was just simply halt the JVM if there was any problem starting up. Logic might have been no point trying to shutdown smoothly if we didn't startup properly. Here the only side effect was possibly leaving kafka/zk processes behind which this PR tries to fix.